darts/models/forecasting_model.py

```diff
@@ -28,54 +32,83 @@ class ForecastingModel(ABC):
 
     @abstractmethod
-    def fit(self) -> None:
+    def fit(self, training_series: TimeSeries) -> None:
         """ Fits/trains the model on the provided series
 
-        Implements behavior that should happen when calling the `fit` method of every forcasting model regardless of
-        wether they are univariate or multivariate.
+        Defines behavior that should happen when calling the `fit()` method of every forecasting model.
+
+        This is the entry point for training the model on one series.
+
+        Some models support training on several time series, or differentiating between "data" and "feature"
+        dimensions. At the moment such functionalities can be used for PyTorch-based models,
+        using the `TimeSeriesTrainDataset`, and calling the `multi_fit()` method.
         """
-        for series in (self.training_series, self.target_series):
-            if series is not None:
-                raise_if_not(len(series) >= self.min_train_series_length,
-                             "Train series only contains {} elements but {} model requires at least {} entries"
-                             .format(len(series), str(self), self.min_train_series_length))
-        self._fit_called = True
+        raise_if_not(len(training_series) >= self.min_train_series_length,
+                     "Train series only contains {} elements but {} model requires at least {} entries"
+                     .format(len(training_series), str(self), self.min_train_series_length))
+
+        self.training_series = training_series
+        self._is_fitted = True
+
+    def multi_fit(self, time_series_train_dataset):
+        raise NotImplemented('This model does not support multi_fit(). Currently only PyTorch-based models'
+                             '(neural nets) support this functionality.')
 
     @abstractmethod
     def predict(self, n: int) -> TimeSeries:
-        """ Predicts values for a certain number of time steps after the end of the training series
+        """ Forecasts values for a certain number of time steps after the end of the training series.
 
         Parameters
         ----------
         n
-            The number of time steps after the end of the training time series for which to produce predictions
+            Forecast horizon - the number of time steps after the end of the series for which to produce predictions.
 
         Returns
         -------
         TimeSeries
-            A time series containing the `n` next points, starting after the end of the training time series
+            A time series containing the `n` next points, starting after the end of the series to forecast.
         """
 
-        if (not self._fit_called):
-            raise_log(Exception('fit() must be called before predict()'), logger)
+        if not self._is_fitted:
+            raise_log(Exception('The model must be fit before calling predict()'), logger)
+
+    def multi_predict(self, n: int, optional_input_series_dataset):
```

Comment:
I guess we have to decide whether the input series should be optional (defaulting to some timestamp from the training set) or not, since earlier in the docstring it says 'The caller of `predict()` has to provide the input series'.